tracer/src/Datadog.Trace/Vendors/IndieSystem.Text.RegularExpressions/common/ValueListBuilder.cs

```diff
@@ -82,12 +86,15 @@ public ReadOnlySpan<T> AsSpan()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Dispose()
         {
+#if !NO_ARRAY_POOL
+            _arrayFromPool = null;
```

Comment:
Is this intended? If it is, the if is useless below, right?

Reply:
Good catch, I think this is missing an `#else` after it @robertpi?

Reply:
I think you're being confused by your own double negatives 😉 

`!NO_ARRAY_POOL` means there _is_ an array pool, which mean `_arrayFromPool` came from the pool, and needs to be returned. But as you're setting `_arrayFromPool = null` here (instead of in the `if` block below) it will never be returned.

Pretty certain you're just missing the `#else` 🙂 
```suggestion
            _arrayFromPool = null;
#else
```

Reply:
you're right, setting the ref to null is an error, but I removed the field in the no-array-pool case, so just deleting the line is enough, done in 24c079227

Thanks you! 🙇‍♂️